### ISSUE TITLE
Add FNV128a

### ISSUE THREAD

**Issue by kokizzu** [Cyan4973/xxHash#714]

**Comment by Cyan4973** [comment]
Is there one canonical version of `FNV128a` that could be employed for testing ?

Also : `FNV128` (non-`a`) was already tested by `rurban/smhasher`, and we already know that the speed is not great and the mixing is not good, particularly for low bits, unsurprisingly, as for all `FNV` variants.
What's expected from another round of tests ?

**Comment by kokizzu** [comment]
I only know this https://github.com/segmentio/fasthash but that one is FNV1a not FNV128a

**Comment by kokizzu** [comment]
this one is FNV128a
https://github.com/deatil/lakego-admin/blob/3a43ae1ec7ad065ed0586a622d4bbd9824666b75/pkg/lakego-pkg/go-hash/hash/fnv.go#L103

apparently it's built-in: https://pkg.go.dev/hash/fnv#New128a

**Comment by Cyan4973** [comment]
> this one is FNV128a https://github.com/deatil/lakego-admin/blob/3a43ae1ec7ad065ed0586a622d4bbd9824666b75/pkg/lakego-pkg/go-hash/hash/fnv.go#L103


Indeed, but it's coded in Go, making it difficult (to put it mildly) to use from a C-based program.

**Comment by kokizzu** [comment]
ah right this is C repo XD my bad
